rootfs/etc/confd/templates/fluent.conf.tmpl

```diff
@@ -224,8 +225,9 @@
   buffer_queue_full_action block
   buffer_chunk_limit {{getv "/fluentd/out/kubesys/buffer/chunk/limit"}}
   buffer_queue_limit {{getv "/fluentd/out/kubesys/buffer/queue/limit"}}
-  flush_interval 5s
+  flush_interval {{getv "/fluentd/out/kubesys/buffer/flush/interval"}}
   max_retry_wait 30
-  disable_retry_limit
+  disable_retry_limit {{getv "/fluentd/out/kubesys/retry/limit/disable"}}
+  retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
```

Comment:
Although my idea would not be that of a well-trained fluentd user, I'd prefer something like:

```
{{if eq (getv "/fluentd/out/kubesys/retry/limit") "-1" -}}
disable_retry_limit true
{{else -}}
retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
{{end -}}
```

so that, imho, the configuration becomes a bit more idiomatic. I've seen a retry limit being "-1" meaning "retry forever" in other job queues and libraries.

Reply:
Thanks for your comment!

>the configuration becomes a bit more idiomatic
I agreed, it's a good!!

I do not understand how to write confd template,,,  I'd like to setting conf like this:

```
{{if le (getv "/fluentd/out/kubesys/retry/limit") 0 -}}
disable_retry_limit true
{{else -}}
retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
{{end -}}
```

Is it possible?

Reply:
While watching [this page](https://github.com/kelseyhightower/confd/blob/master/docs/templates.md), I came up with something  like:

```
{{if exists "/fluentd/out/kubesys/retry/limit" }}
retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
{{else}}
disable_retry_limit true
{{end}}
```
How do you think ? 😄

Reply:
@cw-sakamoto How do you "unset" the retry/limit var to a value other than the default 50 then? If that's possible, it looks good to me :+1:

Reply:
@cw-sakamoto How do you "unset" the retry/limit var to a value other than the default 50 then? If that's possible, it looks good to me :+1:

Reply:
@mumoshu 
Humm...I don't think "unset" this var 😮  Sorry, it was just idea...
Thanks for your considering 👍 

I thought about various cases, but after all I think that the original writing style is good.

so that:
- fluentd user doesn't get confused when setting parameters of kube-fluentd according to fluentd's original specification
  - fluentd user (maybe) knows ```disable_retry_limit``` is boolean value, and disable ```retry_limit``` parameter.
  - When user want to set ```diable_retry_limit```, the same as fluentd is easier to set this parameter.

Reply:
Concretely speaking, perhaps what you couldn't agree with was the "set retry/limit -1 to retry forever" part of my suggestion, right?

Then, I'd suggest:

```
{{if exists "/fluentd/out/kubesys/retry/forever" }}
retry_forever
{{else}}
retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
{{end}}
```

Reply:
>Concretely speaking, perhaps what you couldn't agree with was the "set retry/limit to -1 for retrying forever" part of my suggestion, right?

Yes, I think your suggest becomes configuration more idiomatic and it is important.
But I'm concerned that user could get confused because kube-fluentd's parameter specification is different fluentd's original parameter one.

How do you think my suggestion? 
```
{{if (getv "/fluentd/out/kubesys/retry/forever")}}
retry_forever
{{else}}
retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
{{end}}
```

so that:
- ```"/fluentd/out/kubesys/retry/forever"``` is set to default ```false``` in Dockerfile
- if user want to use retry_forever, user set ```"/fluentd/out/kubesys/retry/forever"``` is other than ```false```
  - ```"/fluentd/out/kubesys/retry/limit"``` is determined if it is valid depending on ```"/fluentd/out/kubesys/retry/forever"```
    - this specification is almost the same to fluentd original one

But, retry_forever is 0.14 only parameter, and retry_limit is 0.12 parameter(retry_limit is converted to retry_max_times internally)... So I'll rewrite with 0.14 parameters.

Please wait ... 😃

Reply:
@cw-sakamoto Thanks for the suggestion 👍 

> if user want to use retry_forever, user set "/fluentd/out/kubesys/retry/forever" is other than false

Perhaps I'm missing something here but does it really work like that?

> ```
> {{if (getv "/fluentd/out/kubesys/retry/forever")}}
> retry_forever
> {{else}}
> retry_limit {{getv "/fluentd/out/kubesys/retry/limit"}}
> {{end}}
> ```

AFAIK, `getv` returns a string value and therefore, won't it always default to `{{if "false"}}` which would be considered `true` in practice? 

> So I'll rewrite with 0.14 parameters.

Great 👍

Reply:
Thanks for your comment.

>AFAIK, getv returns a string value and therefore, won't it always default to {{if "false"}} which would be considered true in practice?

Sorry, it was collect as you said.  😿  I checked my local.
```"false"``` is not boolean, so condition is true. Trying it elsewhare, ```""``` is false and other(```"0"```, ```"false"```) is true.


I'll think about this problem while rewriting for 0.14.